qiskit/quantum_info/operators/symplectic/pauli_list.py

```diff
@@ -177,19 +181,22 @@ def _from_paulis(data):
         if not isinstance(data, (list, tuple, set, np.ndarray)):
             data = [data]
         num_paulis = len(data)
-        if num_paulis == 0:
-            raise QiskitError("Input Pauli list is empty.")
         paulis = []
         for i in data:
             if not isinstance(i, Pauli):
                 paulis.append(Pauli(i))
             else:
                 paulis.append(i)
-        num_qubits = paulis[0].num_qubits
+        if num_qubits is None:
+            if num_paulis == 0:
+                raise QiskitError("Input Pauli list is empty; cannot infer num_qubits.")
+            num_qubits = paulis[0].num_qubits
         base_z = np.zeros((num_paulis, num_qubits), dtype=bool)
         base_x = np.zeros((num_paulis, num_qubits), dtype=bool)
         base_phase = np.zeros(num_paulis, dtype=int)
         for i, pauli in enumerate(paulis):
+            if pauli.num_qubits != num_qubits:
+                raise ValueError("num_qubits of Pauli does not match.")
```

Comment:
we might want to say something like "explicit `num_qubits` does not match the number of Pauli objects given" to make it a little clearer?

Reply:
Actually, `num_qubits` here could be _either_ explicitly provided or inferred from the first element (`paulis[0]`).

In any case, I am going to remove this particular change from this PR (as soon as I push some updates) as it is orthogonal to the goal of the current PR and deserves its own discussion.  I opened #9779, instead, to pursue this.

In that PR, I did try to make an effort for the error message to make sense whether `num_qubits` is provided either implicitly (which is the only option before this current PR) or if it is provided explicitly (as the current PR will make possible).  However, we _could_ consider in _this_ PR that if `i == 0`, we know that `num_qubits` was provided explicitly and so we could consider having an even more informative error message in that case.